src/libs/Navigation/getTopMostCentralPaneRouteName.js

```diff
@@ -0,0 +1,32 @@
+import lodashFindLast from 'lodash/findLast';
+
+/**
+ * Find the name of top most central pane route.
+ *
+ * @param {Object} state - The react-navigation state
+ * @returns {String | undefined} - It's possible that there is no central pane in the state.
+ */
+function getTopMostCentralPaneRouteName(state) {
+    if (!state) {
+        return;
+    }
+    const topmostCentralPane = lodashFindLast(state.routes, (route) => route.name === 'CentralPaneNavigator');
+
+    if (!topmostCentralPane) {
+        return;
+    }
+
+    if (topmostCentralPane.state && topmostCentralPane.state.routes) {
+        // State may don't have index in some cases. But in this case there will be only one route in state.
```

Comment:
The state index should point to the last element in the routes array. But it looks like it's not always the truth. Sometimes the index is not calculated yet. If I remember correctly this can be seen when using getActionFromState

Reply:
> But it looks like it's not always the truth. Sometimes the index is not calculated yet.

hmm this looks quite tricky. anyway, thank you for the explanation. If we're sure that there is only a single route if the `state.index` is not set, I agree we can safely access the first element in the `routes` array.